File: app/src/main/java/com/worldventures/dreamtrips/modules/trips/model/TripClusterItem.java
package com.worldventures.dreamtrips.modules.trips.model;

import com.google.android.gms.maps.model.LatLng;
import com.google.maps.android.clustering.ClusterItem;

public class TripClusterItem implements ClusterItem {

   private Pin pin;

   public TripClusterItem(Pin pin) {
      this.pin = pin;
   }

   @Override
   public LatLng getPosition() {
      return new LatLng(pin.getCoordinates().getLat(), pin.getCoordinates().getLng());
   }

   public Pin getPin() {
      return pin;
   }
}

File: app/src/main/java/com/worldventures/dreamtrips/modules/trips/model/map/TripClusterItem.kt
package com.worldventures.dreamtrips.modules.trips.model.map

import com.google.android.gms.maps.model.LatLng
import com.google.maps.android.clustering.ClusterItem

data class TripClusterItem(val pin: Pin) : ClusterItem {

   override fun getPosition() = LatLng(pin.coordinates.lat, pin.coordinates.lng)
}
